packages/cdk-ecs-codedeploy-resources/lib/index.ts

```diff
@@ -0,0 +1,288 @@
+import * as cdk from "@aws-cdk/core";
+import * as customresource from "@aws-cdk/custom-resources";
+import * as iam from "@aws-cdk/aws-iam";
+import * as s3 from "@aws-cdk/aws-s3";
+import * as cloudfront from "@aws-cdk/aws-cloudfront";
+
+export interface CdkEcsCodedeployResourcesProps {
+  stage: string;
+  clusterName: string;
+  serviceName: string;
+  appName: string;
+  listenerArn: string;
+  targetGroupName: string;
+  commitHash: string;
+}
+
+export class CdkEcsCodedeployResources extends cdk.Construct {
+
+  constructor(scope: cdk.Construct, id: string, props: CdkEcsCodedeployResourcesProps) {
+    super(scope, id);
+
+    const customResourceIamRole = new iam.Role(this, "CustomResourceRole", {
```

Comment:
Both of these roles grant too much. We generally want to provide the least amount of privilege that it needs.